File: Leo Kim/2023-01-11-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        seen = {}

        for k, v in enumerate(nums):
            diff = target - nums[k]

            if diff in seen:
                return [k, seen[diff]]
            else:
                seen[v] = k

File: Dave Min/2023-01-11-TwoSum.java
import java.util.*;
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Hashtable<Integer,Integer> hashtable = new Hashtable<Integer,Integer>(); 
        int num=-1;
        for(int i=0;i<nums.length;i++){
            num = nums[i];
            if(!hashtable.containsKey(num)) {
                hashtable.put(num,i);
            }    
        }
        int pair=0;
        for(int i=0;i<nums.length;i++){             
            num = nums[i];
            pair = target - num;
            if(hashtable.containsKey(pair) && i !=hashtable.get(pair)) 
            return new int[]{i,hashtable.get(pair)};
        }  
        return null;
    }
}